File: test_data/csharp/test_main/aas_core_meta.v3/input/snippets/Verification/ID_shorts_are_unique.cs
/// <summary>
/// Check that all <see cref="Aas.IReferable.IdShort" /> are unique among
/// <paramref name="referables" />.
/// </summary>
public static bool IdShortsAreUnique(
    IEnumerable<Aas.IReferable> referables
)
{
    var idShortSet = new HashSet<string>();
    foreach (var referable in referables)
    {
        if (referable.IdShort != null)
        {
            if (idShortSet.Contains(referable.IdShort))
            {
                return false;
            }
            idShortSet.Add(referable.IdShort);
        }
    }
    return true;
}

File: test_data/java/test_main/aas_core_meta.v3/input/snippets/Verification/ID_shorts_are_unique.java
/**
* Check that all {@link IReferable#getIdShort() idShort}  are unique among referables.
* @param referables the referables.
*/
public static boolean idShortsAreUnique(
        Iterable<IReferable> referables
){
    Set<String> idShortSet = new HashSet<>();
    for (IReferable referable : referables)
    {
        if (referable.getIdShort().isPresent())
        {
            if (idShortSet.contains(referable.getIdShort().get()))
            {
                return false;
            }
            idShortSet.add(referable.getIdShort().get());
        }
    }
    return true;
}
